tests/integration/azure_blob/test_azure_blob.py

```diff
@@ -76,32 +78,57 @@
 class TestAzureBlob:
     @classmethod
     def setup_class(cls):
+        cls.assets_path = os.path.join(
+            os.path.dirname(os.path.dirname(__file__)), "assets"
+        )
         cls.profile_name = "azure_blob_ds_profile"
         cls.test_dir = "test_mlrun_azure_blob"
+        cls.run_dir = cls.test_dir + f"/run_{uuid.uuid4()}"
         cls._bucket_name = config["env"].get("AZURE_CONTAINER", None)
-        cls.test_file = here / "test.txt"
+        cls.test_file = os.path.join(cls.assets_path, "test.txt")
         with open(cls.test_file) as f:
             cls.test_string = f.read()
+        cls._azure_fs = None
 
-    def teardown_method(self):
-        test_dir = f"{self._bucket_name}/{self.test_dir}"
-        if not self._azure_fs:
+    @classmethod
+    def teardown_class(cls):
+        test_dir = f"{cls._bucket_name}/{cls.test_dir}"
+        if not cls._azure_fs:
             return
-        if self._azure_fs.exists(test_dir):
-            self._azure_fs.delete(test_dir, recursive=True)
-            logger.debug("test directory has been cleaned.")
+        if cls._azure_fs.exists(test_dir):
+            cls._azure_fs.delete(test_dir, recursive=True)
+            logger.debug("test directory has been deleted.")
+
+    @classmethod
+    def create_fs(cls, storage_options):
+        #  Create filesystem object only once - for teardown_class method.
+        if not cls._azure_fs:
+            azure_fs = AzureBlobFileSystem(storage_options)
+            try:
+                azure_fs.info(cls._bucket_name)  # in order to check connection ...
+                cls._azure_fs = azure_fs
+            except Exception:
+                logger.warning(
+                    "failed to create azure file system in test_azure_blob.py."
+                )
 
     @pytest.fixture(autouse=True)
     def setup_before_each_test(self, use_datastore_profile, auth_method):
-        self.blob_file = f"file_{uuid.uuid4()}.txt"
+        self.object_file = f"file_{uuid.uuid4()}.txt"
         store_manager.reset_secrets()
-        self._azure_fs = None
         self.storage_options = {}
         for k, env_vars in AUTH_METHODS_AND_REQUIRED_PARAMS.items():
             for env_var in env_vars:
                 os.environ.pop(env_var, None)
 
         test_params = AUTH_METHODS_AND_REQUIRED_PARAMS.get(auth_method)
+        if use_datastore_profile:
+            self._bucket_url = f"ds://{self.profile_name}/{self._bucket_name}"
+        else:
+            self._bucket_url = "az://" + self._bucket_name
+        self.run_dir_url = f"{self._bucket_url}/{self.run_dir}"
+        self.object_url = self.run_dir_url + "/" + self.object_file
```

Comment:
It's a bit inconsistent to use f-string on one line and then not use it in exactly the same situation on the next line.